src/plugins/discover/public/hooks/use_es_doc_search.test.tsx

```diff
@@ -171,23 +170,69 @@ describe('Test of <Doc /> helper / hook', () => {
     `);
   });
 
-  test('useEsDocSearch', async () => {
+  test('useEsDocSearch loading', async () => {
+    const indexPattern = {
+      getComputedFields: () => [],
+    };
+    const props = {
+      id: '1',
+      index: 'index1',
+      indexPattern,
+    } as unknown as DocProps;
+
+    const hook = renderHook((p: DocProps) => useEsDocSearch(p), {
+      initialProps: props,
+      wrapper: ({ children }) => (
+        <KibanaContextProvider services={services}>{children}</KibanaContextProvider>
+      ),
+    });
+
+    expect(hook.result.current.slice(0, 2)).toEqual([ElasticRequestState.Loading, null]);
+  });
+
+  test('useEsDocSearch ignore partial results', async () => {
     const indexPattern = {
       getComputedFields: () => [],
     };
+
+    const record = { test: 1 };
+
     const props = {
       id: '1',
       index: 'index1',
       indexPattern,
     } as unknown as DocProps;
 
-    const { result } = renderHook((p: DocProps) => useEsDocSearch(p), {
+    const hook = renderHook((p: DocProps) => useEsDocSearch(p), {
       initialProps: props,
       wrapper: ({ children }) => (
         <KibanaContextProvider services={services}>{children}</KibanaContextProvider>
       ),
     });
 
-    expect(result.current.slice(0, 2)).toEqual([ElasticRequestState.Loading, null]);
+    await act(async () => {
+      mockSearchResult.next({
+        isPartial: true,
+        isRunning: false,
+        rawResponse: {
+          hits: {
+            hits: [],
+          },
+        },
+      });
+      mockSearchResult.next({
+        isPartial: false,
+        isRunning: false,
+        rawResponse: {
+          hits: {
+            hits: [record],
+          },
+        },
+      });
```

Comment:
dear @Dosant , if you have minute, could you have a look at the unit tests, they work, but I would love to know, if I could make them better ❤️

Reply:
I think that will do it 👍 should capture the flow well. 

Theoretically it is possible to make them behave more real-world by adding delay+fake timers between emits. Maybe even use rxjs-marbles for it. 
But I think that's already good enough for the current use case